all-the-icons-dired.el

```diff
@@ -57,92 +56,43 @@
 
 (defvar all-the-icons-dired-mode)
 
-(defun all-the-icons-dired--add-overlay (pos string)
-  "Add overlay to display STRING at POS."
-  (let ((ov (make-overlay (1- pos) pos)))
-    (overlay-put ov 'all-the-icons-dired-overlay t)
-    (overlay-put ov 'after-string string)))
-
-(defun all-the-icons-dired--overlays-in (beg end)
-  "Get all all-the-icons-dired overlays between BEG to END."
-  (cl-remove-if-not
-   (lambda (ov)
-     (overlay-get ov 'all-the-icons-dired-overlay))
-   (overlays-in beg end)))
-
-(defun all-the-icons-dired--overlays-at (pos)
-  "Get all-the-icons-dired overlays at POS."
-  (apply #'all-the-icons-dired--overlays-in `(,pos ,pos)))
-
-(defun all-the-icons-dired--remove-all-overlays ()
-  "Remove all `all-the-icons-dired' overlays."
-  (save-restriction
-    (widen)
-    (mapc #'delete-overlay
-          (all-the-icons-dired--overlays-in (point-min) (point-max)))))
-
-(defun all-the-icons-dired--refresh ()
-  "Display the icons of files in a dired buffer."
-  (all-the-icons-dired--remove-all-overlays)
-  (save-excursion
-    (goto-char (point-min))
-    (while (not (eobp))
-      (when (dired-move-to-filename nil)
-        (let ((case-fold-search t))
-          (when-let* ((file (dired-get-filename 'relative 'noerror))
-                      (icon (if (file-directory-p file)
-                                (all-the-icons-icon-for-dir file
-                                                            :face 'all-the-icons-dired-dir-face
-                                                            :v-adjust all-the-icons-dired-v-adjust)
-                              (apply 'all-the-icons-icon-for-file file
-                                     (append
-                                      `(:v-adjust ,all-the-icons-dired-v-adjust)
-                                      (when all-the-icons-dired-monochrome
-                                        `(:face ,(face-at-point))))))))
```

Comment:
This is just a personal preference and a minor point: the `(if (file-directory p file) ...)` could be defined as its own function, which accepts a FILE argument.  This makes it easier for the do-it-yourself user to define an override function.